### PR TITLE
Prepare to extract NavigationBar

### PR DESCRIPTION
**Pull request by nickheiner-usds**
image

eFolder will need `NavigationBar`, so we should extract it. In this PR, I'm doing a bit of cleanup. In the second PR, I'll do the actual extraction, and barely change anything else. I think that'll make things easier.

While I was investigating, I also found a bunch of seemingly useless `<div className="cf-app">` wrappers. I removed them. If you view this PR with `?w=1`, you'll see that I didn't changed the content they wrapped.

At Mark's suggestion, I've done some general refactoring of how we show the Caseflow logo. Instead of a bunch of copy/pasted SVG files, we now have a single parameterized React component. (Apps that still use Rails layout for the header, like Certification, will continue to use the old approach.)

Also slipping in [Glamor](https://github.com/threepointone/glamor) like the sly dog I am.

## TODO
Extract the Caseflow icon from `RenderFunctions`, prepare it to be extracted to appeals-frontend-toolkit, and delete the CSS for the icon we currently have.

Delete other unused logo code.

## Other components to extract eventually
* `Footer`
* `LoadingDataDisplay`
* `Table`, but are we sure we can't use a third party component?
* `Button`

### CODE REVIEW

File: client/app/components/NavigationBar.jsx [hunk 1]
```diff
@@ -5,34 +5,11 @@ import Link from './Link';
 import Breadcrumbs from './Breadcrumbs';
 import PerformanceDegradationBanner from './PerformanceDegradationBanner';
 
-const CLASS_NAME_MAPPING = {
```
Comment (nickheiner-usds) [inline client/app/components/NavigationBar.jsx]: I removed this, and shifted this responsibility to the caller. With this approach, we had extra mappings (eFolder, Feedback, Certification, and default), and no clear way to know they weren't used. Also, I find that taking user-facing strings (e.g. `"Hearing Prep"`) and transforming them into identifiers for machines (e.g. `"hearing-prep"`) is not a good practice and does not promote readable code or a flexible UI in the long run.

File: client/app/components/NavigationBar.jsx [hunk 2]
```diff
@@ -43,7 +20,7 @@ export default class NavigationBar extends React.Component {
       <div>
         <div className="cf-app-width">
           <span className="cf-push-left">
-            <h1 className={this.getClassName(appName)}>
+            <h1 className={`cf-logo cf-logo-image-${logoName}`}>
```
Comment (mdbenjam) [inline client/app/components/NavigationBar.jsx]: If we're making this change... What if we moved the SVG from css to code, and just took in the `logoColor` instead? Just a thought.

File: client/app/components/NavigationBar.jsx [hunk 3]
```diff
@@ -4,46 +4,43 @@ import DropdownMenu from './DropdownMenu';
...
       dropdownUrls,
       topMessage,
+      logoProps,
       userDisplayName
     } = this.props;
 
+    const h1Styling = css({
+      margin: 0,
+      display: 'inline-block',
+      'line-height': '3em',
```
Comment (mdbenjam) [inline client/app/components/NavigationBar.jsx]: I think this should be 4em. It looks weird if there are no subheadings:
screen shot 2018-01-12 at 10 40 49 am

File: client/app/components/NavigationBar.jsx [hunk 4]
```diff
@@ -4,46 +4,43 @@ import DropdownMenu from './DropdownMenu';
...
+      fontSize: '1.7rem',
+      '& > a': {
+        color: COLORS.GREY_DARK,
+        paddingLeft: '.3em'
+      }
+    });
+
+    const pushLeftStyling = css({
+      display: 'flex',
```
Comment (mdbenjam) [inline client/app/components/NavigationBar.jsx]: Do we want to introduce flex into all of our apps? IE support is weak: https://caniuse.com/#search=flexbox

File: client/app/establishClaimAdmin/index.jsx [hunk 5]
```diff
@@ -23,6 +23,10 @@ const EstablishClaimAdmin = ({
         <div>
           <NavigationBar
             appName="Establish Claim Admin"
+            logoProps={{
+              accentColor: '#844e9f',
```
Comment (mdbenjam) [inline client/app/establishClaimAdmin/index.jsx]: I tend to think all these should be constants. In fact some of them already are.

File: client/app/components/NavigationBar.jsx (hunk 4 above)
Comment (nickheiner-usds) [reply client/app/components/NavigationBar.jsx]: https://github.com/department-of-veterans-affairs/caseflow/pull/4336#issuecomment-358025235

**Comment by nickheiner-usds** [comment]
> Why does 'Caseflow' disappear sometimes when reloading?

I've observed the same thing on `master`. It particularly caused issues with the Hearings Prep print view page.

I don't know why it happens. I thought that maybe it was a font loading slowly, but the text that shows up immediately is the same font.

**Pull request merged**